packages/babel-parser/test/fixtures/typescript/class/accessor-invalid/output.json

```diff
@@ -0,0 +1,220 @@
+{
+  "type": "File",
+  "start":0,"end":239,"loc":{"start":{"line":1,"column":0,"index":0},"end":{"line":15,"column":1,"index":239}},
+  "errors": [
+    "SyntaxError: An 'accessor' property cannot be declared optional. (7:2)"
```

Comment:
It seems that other errors are not reported.

Reply:
Yes, I only reported that unrecoverable error.
Actually initially I also implemented another one, but then I thought they didn't matter.

Reply:
Doesn't the `output.json` in parser tests only report recoverable errors unless one set `errorRecovery: false`? The unrecoverable error will be reported at `options.throw`.

Reply:
I mean currently the only errors reported are from unrecoverable errors. And the rest are illegal but normal parsing, I don't think it matters, so they haven't been implemented.

Reply:
> An accessor can not be optional. We are currently throwing unexpected token at ?. We can provide a better error here.

https://babeljs.io/repl#?browsers=&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=MYGwhgzhAEDC0G8BQ1pmMAplA9gJ2gGIAPAfgG4kBfIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=stage-2&prettier=false&targets=Node-0&version=7.20.4&externalPlugins=&assumptions=%7B%7D

Reply:
> And the rest are illegal but normal parsing

Although `tsc` parser is a loose parser, it does not necessarily mean it should not report errors. The `errorRecovery` option enables Babel the same tsc functionality: report errors without crashing the parser.

Generally the Babel parser does not report typing errors, such as "the call signature does not match the function interface": they are considered as "runtime errors" because these errors can only be reported from a typing system.

But for those errors that can be detected solely from AST, they are most static errors and thus they should be reported by the parser. We have reported similar errors, e.g. Babel throws "Private elements cannot have an accessibility modifier" for
```ts
class C {
  private #Q;
}
```
Therefore I am not convinced the idea that "the rest are illegal but normal parsing". If TS would ever have a spec, they should be early errors and a parser should report them.

Reply:
I personally prefer to make unrecoverable errors recoverable, and normally parsed ts errors are not actively reported. (at least when no user requests these)
Because I believe all users will use tsc at the same time, and ts makes too many errors recoverable.

> My bad, by a better error message, I do mean that we can recover from this error and throw a recoverable error.

I've done this and the only error is from what you said.